src/kfactory/routing/manhattan.py

```diff
@@ -212,7 +211,169 @@ def droute_manhattan(
+                        # route like a P
+                        if x < 0:
+                            # the straight part of the P is on our side
+                            self.straight(max(2 * self.bend90_radius + x, 0))
+                        if y > 0:
+                            self.right()
+                        else:
+                            self.left()
+                        return self.auto_route(max_try - 1)
+            case 2:
+                match y:
+                    case 0:
+                        return self.finish()
+                    case y if y_abs < 2 * self.bend90_radius:
+                        self.right() if y > 0 else self.left()
+                        return self.auto_route(max_try - 1)
+                    case _:
+                        if y > 0:
+                            self.left()
+                            self.straight(y_abs - 2 * self.bend90_radius)
+                            self.right()
+                        else:
+                            self.right()
+                            self.straight(y_abs - 2 * self.bend90_radius)
+                            self.left()
+                        return self.finish()
+            case _:
+                # 1/3 cases are just one to the other
+                # with flipped y value and right/left flipped
+                if ta == 3:
+                    right = self.right
+                    left = self.left
+                    _y = y
+                else:
+                    right = self.left
+                    left = self.right
+                    _y = -y
+                if x >= self.bend90_radius and _y >= self.bend90_radius:
+                    # straight forward can connect with a single bend
+                    self.straight(x - self.bend90_radius)
+                    left()
+                    return self.finish()
+                if x >= 3 * self.bend90_radius:
+                    # enough space to route but need to first make sure we have enough
+                    # vertical way (seen from t1)
+                    right()
+                    return self.auto_route(max_try - 1)
+                if _y >= 3 * self.bend90_radius:
+                    # enough to route in the other side
+                    left()
+                    return self.auto_route(max_try - 1)
+                if _y <= 0 or x <= 0:
+                    self.straight(max(x + self.bend90_radius, 0))
+                    right()
+                    return self.auto_route(max_try - 1)
+
+                # attempt small routing
+                config.logger.warning(
+                    "route is too small, potential collisions: "
+                    f"{self.t1=}; {self.t2=}; {self.pts=}"
+                )
+
+                right()
+                self.straight(max(self.bend90_radius - _y, 0))
+                left()
+                return self.auto_route(max_try - 1)
```

Comment:
@joamatab this is the whole routing logic. Essentially check whether the ports are parallel or facing each other, otherwise make some biased choices to connect directly or go for a round.